timon/state.py

```diff
@@ -295,38 +295,38 @@ def has_state_changed(self, probe, status, flap_detection=True,
             previous_rslt_to_compare = pst[-flap_cnt]
             return previous_rslt_to_compare[1] != status
 
-    async def update_probe_state(
-            self, probe,
-            status, msg=None, t=None, save=False):
+    async def update_probe_state(self, probe, status, msg, t):
```

Comment:
can we be sure, that update_probe_state is always called with all the  parameters.

timon is theoretically an open source project and could be used by others.

`msg=None`, etc. allows to call the function without  these params.

removing the default values breaks theoretically backwards compatibility.

Reply:
okay, fair; I'll leave the crap like it was then

an API _neeeeds_ to be defined more specifically, 90% of the functions here don't know what they call nor what they do, this one even has a kill switch!

Reply:
also hate to break it to you but this repo has 0 actual forks and 6 active vulnerabilities; but I don't know how much it's used by others...